static/fetch.js

```diff
@@ -38,48 +186,25 @@ class AutoReload {
     }
     async update(last_fetch = 0){
         let fetched = await fetch(`/reload/recent?last-fetch=${last_fetch}`);
-        let data = await fetched.json();
+        let allPrinters = await fetched.json();
         this.last_fetch = Date.now();
         if (!this.printers) {
             this.printers = {};
-            for (let printer_name in data) {
-                this.printers[printer_name] = new Map();
-            }
+            for (let printer_name in allPrinters) this.printers[printer_name] = new Printer(printer_name);
         }
         let currTime = Date.now();
-        for (let printer_name in data) {
-            let parent = document.querySelector(`.${printer_name} h1`);
-            data[printer_name].sort((a, b) => a[1] - b[1])
-            .forEach(item => {
-                let job = new Job(this, printer_name, item[0], item[1]);
-                this.printers[printer_name].set(job.symbol, job);
-                parent.after(job.element);
-            });
-            this.printers[printer_name].forEach(job => {
-                if (currTime - job.date.getTime() > PERSISTENCE_TIME)
-                    job.remove();
-            });
-            if (this.printers[printer_name].size == 0) {
-                if (!document.querySelector(`.${printer_name} .printlist-null`))
-                    parent.after(this.createNull());
-            }
-            else
-                this.deleteNull(printer_name);
+
+        for (let printer_name in allPrinters) {
+            let targetPrinter = this.printers[printer_name];
+            allPrinters[printer_name].sort((a, b) => a.time - b.time)
+            .forEach(rawJob => targetPrinter.addJob(rawJob));
+
+            targetPrinter.clean(currTime);
+            targetPrinter.updateSize();
         }
     }
-    createNull(){
-        let nullDiv = document.createElement('div');
-        nullDiv.classList.add('printlist-null');
-        nullDiv.innerText = '(there are no jobs printing)';
-        return nullDiv;
-    }
-    deleteNull(printer_name){
-        let target = document.querySelector(`.${printer_name} .printlist-null`);
-        if (target)
-            target.remove();
-    }
 }
-
+let debug;
```

Comment:
:thinking:

Reply:
Right, this was for debugging purposes (so you could access the AutoReload object). Should I remove it?